src/literal.h

```diff
@@ -97,13 +97,13 @@ class Literal {
   Literal extendS8() const;
   Literal extendS16() const;
   Literal extendS32() const;
-  Literal truncateToI32() const;
-  Literal truncateToF32() const;
+  Literal wrapToI32() const;
+  Literal demoteToF32() const;
```

Comment:
actually I think we should just remove `demoteToF32`. It's not used, there's a `demote` method which is actually tested, and it has more logic to handle corner cases.

lgtm with that